presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1633,7 +1633,9 @@ public String getValidWriteIds(HiveIdentity identity, List<SchemaTableName> tabl
                     }));
         }
         catch (TException e) {
-            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+            // When calling Hive metastore < 3, the call fails with
+            // Required field 'open_txns' is unset! Struct:GetOpenTxnsResponse(txn_high_water_mark:4, open_txns:null, min_open_txn:4, abortedBits:null)
```

Comment:
should we verify if it is the case before raising an exception with below message. 

In case of other communication errors it could be misleading.

Reply:
This is an individual call, so there is small risk of mis-handling it, if all previous calls worked.